sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -232,14 +209,14 @@ class StatefulProcessorHandleImpl(
    * @param stateName - name of the state variable
    */
   override def deleteIfExists(stateName: String): Unit = {
-    verifyStateVarOperations("delete_if_exists")
+    verifyStateVarOperations("delete_if_exists", CREATED)
```

Comment:
Why do we pass `CREATED` here ?

Reply:
In DriverStatefulProcessorHandleImpl, we expect the state to be `PRE_INIT`, and here, it must be `CREATED`, so we pass in the expected state into the `verifyStateVarOperations` method so we can share it.